### DISCUSSION TITLE
New millipds users are not able to federate with bsky appview

### DISCUSSION THREAD

**Discussion by DavidBuchanan314** [bluesky-social/atproto#3215] [Dev Debugging] [answered]
Around the 17th of November, I set up an instance of [millipds](https://github.com/DavidBuchanan314/millipds) at https://pds.dev.retr0.id/

Everything worked great (mostly), and the account is visible at https://bsky.app/profile/david.dev.retr0.id

Over the past few days, a few other people have set up millipds instances:

- https://biden.lgbt (cc @OkashiOdayakana )
    - https://bsky.app/profile/did:plc:beq5fpl2yml5lzlr2nlmmbzl
- https://pds.dev.goranmoomin.dev (cc @goranmoomin )
    - https://bsky.app/profile/did:plc:ilbuexsezfgfn6ospjcpkupt
- https://pds.dev.lily.pet (cc @lilyvxv )
    - https://bsky.app/profile/did:plc:h5f4kq235u7lvpe2gf5ptyom

Unfortunately, they've not had so much luck. I can't debug very effectively from my vantage point, but the first two accounts in particular appear to be connecting to the relay correctly, just not the appview - none of their profiles or posts are visible in bsky.app.

The issue may well be on millipds's end, but it's hard to debug further without knowing what's going on inside the appview - I'm hoping we might be able to get some insight there.

Of course, I'm happy to answer questions about "what millipds does", along with my setup for `david.dev.retr0.id` - and I'm hoping the users I've tagged are available to answer questions about their own setups.

I recently wrote up instructions for setting up a new account on millipds, which they may have been following: https://github.com/DavidBuchanan314/millipds/blob/main/docs/ACCOUNTS.md

**Comment by DavidBuchanan314** [comment]
Possibly related: https://github.com/bluesky-social/atproto/discussions/3214 "Accounts on federated PDSes sometimes get "stuck" and stop updating in the appview"

(although in this case, they never worked in the first place)

**Comment by DavidBuchanan314** [comment]
Something that may be related - millipds does not yet properly set the `didDoc` field in response to `com.atproto.repo.describeRepo` queries (it just sets it to `{}`)

**Comment by devinivy** [comment]
Booting-up millipds to take a look.

**Reply by devinivy** [reply]
As far as I can tell millipds is working just fine (also very fun to hack around with).  I think I've narrowed it down to an issue related to backfill behaviors in the Bluesky appview, digging into it!

**Reply by devinivy** [reply, marked as answer]
@DavidBuchanan314 I think we got to the bottom of this, and those profiles linked above appear to be indexed now.  My millipds setup is also being indexed again normally.  My hunch about "what's different" about millipds that triggered the behavior is that it may have to do with either a. not having a commit event for the repo init (just a commit block), or b. emitting some events prior to the first account/identity events for the repo.  Both should have been things that we were recovering from—and I believe have been addressed—but may be why the same thing wasn't happening on the mushroom servers.

**Reply by DavidBuchanan314** [reply]
Thanks for looking into this! So everything millipds currently does wrt account activation is at least "not incorrect"?

I definitely plan on automating the account/identity events in the future (so they happen at more correct times), and I can think about emitting a "commit zero" that commits the initial empty MST
